### ISSUE TITLE
Test runs fine individually but fails when "run all" is given

### ISSUE THREAD

**Issue by srinivasrk** [cypress-io/cypress#2028]
### Current behavior:
I am uploading a csv file using XHR and form data. It takes nearly a minute for the server to parse the .csv and write the content in the database. These test cases work fine when run by itself, when I click on run all it fails and it gets status 500 even when the server sends back 200 !

![image](https://user-images.githubusercontent.com/11569731/41868807-2dd6370a-7885-11e8-9da3-e7c41a382b8e.png)

![image](https://user-images.githubusercontent.com/11569731/41869032-ad319120-7885-11e8-99a1-fe667cb827b3.png)

![image](https://user-images.githubusercontent.com/11569731/41868910-636cba24-7885-11e8-82a6-be4f5b643188.png)

![image](https://user-images.githubusercontent.com/11569731/41869003-9eef43a0-7885-11e8-8e62-3415d0d3d134.png)

**Comment by srinivasrk** [comment]
I guess it was timing out. I increased to the time out to work.

```
    it('should verify the file upload for channels', function() {
      const fileName = 'channels.csv';
      const method = 'POST';
      const url = 'http://localhost:3000/api/channels';
      const fileType = 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet';
      const expectedAnswer = 'Successfully updated channels database';

      // Get file from fixtures as binary
      cy.fixture(fileName, 'binary').then((excelBin) => {
          // File in binary format gets converted to blob so it can be sent as Form data
          Cypress.Blob.binaryStringToBlob(excelBin, fileType).then((blob) => {
              // Build up the form
              const formData = new FormData();
              formData.set('file', blob, fileName); //adding a file to the form
              // Perform the request
              cy.form_request(method, url, formData, function (response) {
                  console.log(response);
                  expect(response.status).to.equal(200)
                  expect(response.response).to.equal(expectedAnswer)
              })
          })
      }).wait(55000)
    });
```

**Comment by brian-mann** [comment]
You're not returning the `Cypress.Blog.binaryStringToBlob` promise which is going to create a race condition that can cause random failures / timeouts, etc.

You have to return promises to a `cy.then` so it knows to await it, else it has no idea you've kicked one off and it doesn't know when to wait to resolve.

Also - running "All Tests" from the GUI is not a good idea. We've documented this in detail in another issue, which is why we moved to running all of your specs in isolation with `cypress run`.

In other words, don't use `cypress open` to run all your test. Use `cypress run` to do it. Yes, I know we do provide that button, and we've internally talked about how to improve this, but the work isn't done yet, so we've left things as is to keep current users happy.

- https://github.com/cypress-io/cypress/issues/681
- https://github.com/cypress-io/cypress/issues/1586

**Comment by srinivasrk** [comment]
Thank you @brian-mann

**Comment by jcn-ithaca** [comment]
@brian-mann : You wrote that 'running "All Tests" from the GUI is not a good idea';
Is this still true in Cypress 4.6? 
We're experiencing tests that succeed individually but fail during 'Run All'.
thanks
